testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -822,6 +822,42 @@ public void check3pCookiesMessageCallbackTest() {
                 .init(defaultArguments(), this::assertInitNotAuth);
     }
 
+    // In case of incorrect/unavailable realm provided in KeycloakConfig,
+    // JavaScript Adapter init() should fail-fast and reject Promise with KeycloakError.
+    @Test
+    public void checkInitWithInvalidRealm() {
+
+        JSObjectBuilder keycloakConfig = JSObjectBuilder.create()
+                .add("url", authServerContextRootPage + "/auth")
+                .add("realm", "invalid-realm-name")
+                .add("clientId", CLIENT_ID);
+
+        JSObjectBuilder initOptions = defaultArguments();
+
+        testExecutor
+                .configure(keycloakConfig)
+                .init(initOptions, this::assertErrorResponse);
```

Comment:
We should check also error response here:
```suggestion
                .init(initOptions, assertErrorResponse("Realm does not exist"));
```